srcts/src/components/submitTextArea.ts

```diff
@@ -0,0 +1,209 @@
+import type { InputSubscribeCallback } from "rstudio-shiny/srcts/types/src/bindings/input/inputBinding";
+import { registerBinding, InputBinding, updateLabel } from "./_utils";
+
+type TextSubmitReceiveMessageData = {
+  value?: string;
+  placeholder?: string;
+  label?: string;
+  submit?: boolean;
+  focus?: boolean;
+};
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const EVENT_NAMESPACE = "textSubmitInputBinding";
+
+// When a textarea becomes visible, update the height
+const intersectObserver = new IntersectionObserver((entries) => {
+  entries.forEach((entry) => {
+    if (entry.isIntersecting) {
+      updateHeight(entry.target as HTMLTextAreaElement);
+    }
+  });
+});
+
+class TextAreaSubmitInputBinding extends InputBinding {
+  #submitButton: HTMLButtonElement | null = null;
+
+  find(scope: HTMLElement): JQuery<HTMLElement> {
+    return $(scope).find(".bslib-input-textsubmit > textarea");
+  }
+
+  initialize(el: HTMLTextAreaElement): void {
+    const btn = el.nextElementSibling;
+    // This assumption is forced server-side
+    if (!(btn instanceof HTMLButtonElement)) {
+      throw new Error("No submit button found");
+    }
```

Comment:
I'd rather we add a class to the button. These positional assumptions always make me nervous and it unnecessarily couples the R and JS logic, which makes it more likely that we break something by accident if we do any kind of moving things around.

The class approach is also nice because it gives users something direct to work with to style the text input submit button rather than also relying on positional selectors that we could accidentally break in the future.